cbits/client.c

```diff
@@ -105,8 +106,14 @@ static int assert_connection(void)
 bool await_go(void)
 {
     char buf[2];
-    ssize_t rc = recv(assert_connection(), PS(buf), 0);
-    return 2 == rc && !memcmp("GO", PS(buf));
+    uint32_t left = sizeof(buf);
+    while (left > 0) {
+        ssize_t rc = recv(assert_connection(), PS(buf), 0);
```

Comment:
ugh, yes. I guess what's happening in practice is that it's getting EINTR before the first byte arrives, then it gets both bytes.